nifi-docker/dockerhub/Dockerfile

```diff
@@ -16,37 +16,37 @@
 # under the License.
 #
 
-FROM openjdk:8
-MAINTAINER Apache NiFi <dev@nifi.apache.org>
+FROM openjdk:8-jre
+LABEL maintainer "Apache NiFi <dev@nifi.apache.org>"
 
 ARG UID=1000
-ARG GID=50
 ARG NIFI_VERSION=1.4.0
+ARG MIRROR=https://archive.apache.org/dist
 
-ENV NIFI_BASE_DIR /opt/nifi
-ENV NIFI_HOME $NIFI_BASE_DIR/nifi-$NIFI_VERSION
-ENV NIFI_BINARY_URL https://archive.apache.org/dist/nifi/$NIFI_VERSION/nifi-$NIFI_VERSION-bin.tar.gz
+ENV NIFI_BASE_DIR /opt/nifi 
+ENV NIFI_HOME=$NIFI_BASE_DIR/nifi-$NIFI_VERSION \
+    NIFI_BINARY_URL=/nifi/$NIFI_VERSION/nifi-$NIFI_VERSION-bin.tar.gz
 
 # Setup NiFi user
-RUN groupadd -g $GID nifi || groupmod -n nifi `getent group $GID | cut -d: -f1`
-RUN useradd --shell /bin/bash -u $UID -g $GID -m nifi
-RUN mkdir -p $NIFI_HOME
+RUN groupadd nifi && useradd nifi --shell /bin/bash -u $UID -m -g nifi  \
+    && mkdir -p $NIFI_HOME/conf/templates && chown -R nifi:nifi $NIFI_BASE_DIR
+
+USER nifi
 
 # Download, validate, and expand Apache NiFi binary.
-RUN curl -fSL $NIFI_BINARY_URL -o $NIFI_BASE_DIR/nifi-$NIFI_VERSION-bin.tar.gz \
-	&& echo "$(curl $NIFI_BINARY_URL.sha256) *$NIFI_BASE_DIR/nifi-$NIFI_VERSION-bin.tar.gz" | sha256sum -c - \
+RUN curl -fSL $MIRROR/$NIFI_BINARY_URL -o $NIFI_BASE_DIR/nifi-$NIFI_VERSION-bin.tar.gz \
+	&& echo "$(curl https://archive.apache.org/dist/$NIFI_BINARY_URL.sha256) *$NIFI_BASE_DIR/nifi-$NIFI_VERSION-bin.tar.gz" | sha256sum -c - \
 	&& tar -xvzf $NIFI_BASE_DIR/nifi-$NIFI_VERSION-bin.tar.gz -C $NIFI_BASE_DIR \
-	&& rm $NIFI_BASE_DIR/nifi-$NIFI_VERSION-bin.tar.gz
+	&& rm $NIFI_BASE_DIR/nifi-$NIFI_VERSION-bin.tar.gz \
+    && chown -R nifi:nifi $NIFI_HOME
 
-RUN chown -R nifi:nifi $NIFI_HOME
+# Web HTTP Port & Remote Site-to-Site Ports
+EXPOSE 8080 8181
 
-# Web HTTP Port
-EXPOSE 8080
+WORKDIR $NIFI_HOME
 
-# Remote Site-To-Site Port
-EXPOSE 8181
-
-USER nifi
+VOLUME conf/templates
 
```

Comment:
Can that be an extended discussion elsewhere? While I might agree, at least for nifi.properties, the primary goal of this PR was to cut down the image size.